federation.go

```diff
@@ -24,19 +24,83 @@ type FederationDefinition struct {
 
 // Represents a configured Federation upstream.
 type FederationUpstream struct {
+	Name       string               `json:"name"`
+	Vhost      string               `json:"vhost"`
+	Component  string               `json:"component"`
+	Definition FederationDefinition `json:"value"`
+}
+
+// FederationDefinitionDTO provides a data transfer object for a FederationDefinition.
+type FederationDefinitionDTO struct {
 	Definition FederationDefinition `json:"value"`
 }
 
+//
+// GET /api/parameters/federation-upstream
+//
+
+// ListFederationUpstreams returns all federation upstreams
+func (c *Client) ListFederationUpstreams() (rec []FederationUpstream, err error) {
```

Comment:
It would be very nice to accompany this (and, perhaps, base it on)  with a set of functions that would list, set, and delete arbitrary runtime parameters. Note that RabbitMQ nodes will perform parameter validation so if a component is not known (not registered by a plugin or RabbitMQ core), it will be rejected.

We can also add functions that work with parameters of a certain type. Then federation upstream management would rely on them and pass `federation-upstream` for component. How does this sound?